West Midlands | ITP-MAY-25 | Abdullah Saleh | Form control 

3bdullah-saleh opened the pull request:
<!--

You must title your PR like this:

REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME

For example,

London | May-2025 | Carol Owen | Wireframe

Complete the task list below this message.
If your PR is rejected, check the task list.

-->

## Learners, PR Template

Self checklist

- [x] I have committed my files one by one, on purpose, and for a reason
- [x] I have titled my PR with REGION | COHORT_NAME | FIRST_NAME LAST_NAME | PROJ_NAME
- [x] I have tested my changes
- [x] My changes follow the [style guide](https://curriculum.codeyourfuture.io/guides/reviewing/style-guide/)
- [x] My changes meet the [requirements](./README.md) of this task

## Changelist

This PR completes the core functionality and styling of the T-shirt order form as part of the form controls exercise.
## Questions

Ask any questions you have for your reviewer.

cjyuan reviewed (commented):
1. According to W3C Markup Validation Service, there are some warnings and errors in your code. Can you fix them?

2. Currently your form has White and XS as the pre-selected choices. Can you find out from ChatGPT the pros and cons of making a default-selected radio button, and decide whether your form should (or should not) pre-select the T-shirt color and size?

3. Can you also run your code through ChatGPT to see if there’s anything (no matter how minor) that could be improved? I noticed some small inconsistencies, but I believe you can identify them yourself with ChatGPT’s help.

3bdullah-saleh commented:
I updated the form:

Replaced radio buttons with <select> for color and size inputs

Improved name field validation to prevent empty space submissions

Removed external CSS as per the spec

Fixed all W3C validation issues

cjyuan commented on Form-Controls/index.html:
```diff
@@ -11,17 +11,108 @@
+            <option value="black">Black</option>
+          </select>
+        </fieldset>
+        <!-- <div>
+          <fieldset>
+            <legend>What color should the T-shirt be?</legend>
+            <ul>
+              <li>
+                <input
+                  type="radio"
+                  name="color"
+                  id="white"
+                  value="white"
+                  required
+                />
+                <label for="white">White</label>
+              </li>
+              <li>
+                <input type="radio" name="color" id="blue" value="blue" />
+
+                <label for="blue">Blue</label>
+              </li>
+              <li>
+                <input type="radio" name="color" id="black" value="black" />
+
+                <label for="black">Black</label>
+              </li>
+            </ul>
+          </fieldset>
+        </div> -->
```
To keep a PR as clean as possible, we should not include any unused code in the PR. Do you need these code for the form implementation?

cjyuan commented on Form-Controls/index.html:
```diff
@@ -11,17 +11,108 @@
+        </div> -->
+        <br />
+        <fieldset>
+          <legend>What size should the T-shirt be?</legend>
+          <label for="size">Size</label>
+          <select id="size" name="size" required>
+            <option  value="" disabled selected>Select a size</option>
```
Can you compare lines 97-100 lines to lines 59-62, and improve the consistency of letter casing in the text content of these two `<select>` elements?

cjyuan replied:
The text in the label still have some inconsistency.

cjyuan replied:
Yes.

cjyuan commented on Form-Controls/index.html:
```diff
@@ -11,17 +11,108 @@
     </header>
+    <section aria-label="Required field note">
+      <p>
+        Required fields are marked with
+        <strong>
+          <span aria-hidden="true">*</span>
+        </strong>
```
Not all required fields are currently marked with *.

cjyuan reviewed (commented):
Changes look good.

Can you address the few comments I left with the code (regarding consistency and unused code)?

3bdullah-saleh commented:
I removed the comments, added asterisks to required fields, and standardized letter casing in dropdown

cjyuan commented:
Changes look good. 

Please note that replying directly to inline comments in the PR can help keep discussions organized and easier to resolve.
![image](https://github.com/user-attachments/assets/8035eeb6-99af-4696-8eae-f3d440ac8880)
